### PR TITLE
Module skeleton

### PR DESCRIPTION
**Pull request by nottrobin**

### CODE REVIEW

**Review by WillMoggridge — APPROVED** [review]

**Comment by tbille** [comment]
Since it's a new project could we integrate CircleCI? (maybe just the linter for now?)

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#! /usr/bin/env python3
+
+# Core
+from setuptools import setup, find_packages
+
+setup(
+    name='canonicalwebteam.http',
```
Comment (squidsoup) [inline setup.py:7]: The package name should just be called `canonicalwebteam`, which also suggests that this repo should be called `canonicalwebteam`. `http` is just a module within the `canonicalwebteam` package.

Comment (nottrobin) [reply setup.py:7]: There [is a (possibly somewhat loose) recommendation](https://legacy.python.org/dev/peps/pep-0423/#top-level-namespace-relates-to-code-ownership) to use namespaces in PyPi package names to denote ownership - `<owner>.<package>`

We therefore use `canonicalwebteam` as a namespace, for collecting our PyPi packages - e.g. [canonicalwebteam.gsa](https://github.com/canonical-webteam/gsa), [canonicalwebteam.yaml-redirect](https://github.com/canonical-webteam/yaml-redirects) etc. These can exist in harmony through [packaging namespace packages](https://packaging.python.org/guides/packaging-namespace-packages/), and for that pattern you don't mention the namespace module in `setup.py`, only the child module.

I would rather not collect all these packages together in a single project or codebase, as they make much more sense and are easier to understand and use in isolation.

**Comment by squidsoup** [comment]
Ah interesting, I must admit that's the first time I've seen that. Thanks
Robin.

**Comment by nottrobin** [comment]
I tried to get CircleCI working, but I think it might be easier if I did it directly on Master. So I'll merge this now and continue to experiment with CircleCI.

**Pull request merged**